src/detect-engine-tag.c

```diff
@@ -648,42 +647,41 @@ static int DetectTagTestPacket01 (void)
     HostInitConfig(1);
 
     SCLogDebug("running tests");
-    result = UTHGenericTest(p, 7, sigs, sid, (uint32_t *) results, 5);
+    FAIL_IF_NOT(UTHGenericTest(p, 7, sigs, sid, (uint32_t *)results, 5));
     SCLogDebug("running tests done");
 
     Host *src = HostLookupHostFromHash(&p[1]->src);
+    FAIL_IF_NULL(src);
     if (src) {
         void *tag = HostGetStorageById(src, host_tag_id);
         if (tag != NULL) {
             printf("tag should have been expired: ");
-            result = 0;
         }
-
+        FAIL_IF_NOT_NULL(tag);
         HostRelease(src);
     }
     Host *dst = HostLookupHostFromHash(&p[1]->dst);
-    if (dst) {
-        void *tag = HostGetStorageById(dst, host_tag_id);
-        BUG_ON(tag == NULL);
+    FAIL_IF_NULL(dst);
 
-        DetectTagDataEntry *iter = tag;
+    void *tag = HostGetStorageById(dst, host_tag_id);
+    FAIL_IF_NULL(tag);
 
-        /* check internal state */
-        if (!(iter->gid == 1 && iter->sid == 2 && iter->packets == 4 && iter->count == 4)) {
-            printf("gid %u sid %u packets %u count %u: ", iter->gid, iter->sid, iter->packets, iter->count);
-            result = 0;
-        }
+    DetectTagDataEntry *iter = tag;
 
-        HostRelease(dst);
+    /* check internal state */
+    if (!(iter->gid == 1 && iter->sid == 2 && iter->packets == 4 && iter->count == 4)) {
+        printf("gid %u sid %u packets %u count %u: ", iter->gid, iter->sid, iter->packets,
+                iter->count);
     }
-    BUG_ON(src == NULL || dst == NULL);
```

Comment:
I'm not sure these are the right way of approaching this task. I'm taking a better look at this.

Reply:
@jufajardini
Do you have any insight?

Reply:
Am  I missing the whole point to the test?

Reply:
I'm sorry, I had emergencies that didn't let me finish reviewing this yesterday. I think some checks and statements may be missing, that's what I was trying to investigate yesterday. 

My understanding of this conversion is that the `if` statements in the unit tests can all be avoidable/converted to the new FAIL/PASS API. But sometimes some checks are more convoluted, and therefore it gets harder to understand how to convert them.

Reply:
I just checked this test, and it seems possible to convert it removing/converting all if statements.

It also seems ok to remove the `BUG_ON` statements here, given that we'll have `FAIL_IF_NULL` checks for those cases...